android/src/main/kotlin/com/signify/hue/flutterreactiveble/ble/ReactiveBleClient.kt

```diff
@@ -37,15 +37,20 @@ open class ReactiveBleClient(private val context: Context) : BleClient {
     private val connectionQueue = ConnectionQueue()
     private val allConnections = CompositeDisposable()
 
-    override val connectionUpdateSubject: BehaviorSubject<ConnectionUpdate> = BehaviorSubject.create()
-
     companion object {
+        // this needs to be in companion update since backgroundisolates respawn the eventchannels
+        // Fix for https://github.com/PhilipsHue/flutter_reactive_ble/issues/277
+        private val connectionUpdateBehaviorSubject: BehaviorSubject<ConnectionUpdate> =
+            BehaviorSubject.create()
```

Comment:
There are multiple approaches to support multiple Flutter engines. Doing that is definitely a very good idea, but is this the only change needed?

Reply:
> There are multiple approaches to support multiple Flutter engines. Doing that is definitely a very good idea, but is this the only change needed?

Yes I did check the rest of the operations and this is the only one that is needed since they are either linked to the same connection subject or they will be re initialized due to the fact we use embedding v2